src/js/components/TaskDebugView.js

```diff
@@ -1,24 +1,70 @@
 import classNames from 'classnames';
 import {Dropdown} from 'reactjs-components';
 import React from 'react';
+import ReactDOM from 'react-dom';
 
+import FilterBar from './FilterBar';
 import FilterInputText from './FilterInputText';
 import IconDownload from './icons/IconDownload';
 import MesosLogView from './MesosLogView';
 import TaskDirectoryActions from '../events/TaskDirectoryActions';
 
-const METHODS_TO_BIND = ['handleSearchStringChange'];
+const METHODS_TO_BIND = [
+  'handleSearchStringChange',
+  'handleCountChange',
+  'handleKeyDown'
+];
 
 class TaskDebugView extends React.Component {
   constructor() {
     super();
 
-    this.state = {currentFile: null};
+    this.state = {
+      currentFile: null,
+      watching: 0,
+      totalFound: 0
+    };
 
     METHODS_TO_BIND.forEach((method) => {
       this[method] = this[method].bind(this);
     });
+  }
+
+  componentDidMount() {
+    let filterInput = ReactDOM.findDOMNode(this.refs.filterInput);
+
+    if (filterInput) {
+      filterInput.addEventListener('keydown', this.handleKeyDown);
+    }
+  }
+
+  componentWillReceiveProps(nextProps, nextState) {
+    let {searchString, totalFound} = nextState;
+    let updatedState = {};
+
+    if (totalFound === 0) {
+      updatedState.watching = 0;
+    } else if (this.state.watching === 0 ||
+      this.state.searchString !== searchString && searchString != null) {
```

Comment:
1. Let's check if `searchString` is defined before we try to determine if it is not equal to `this.state.searchString`.
2. Maybe we could also define better variables? Usually in the `componentWillReceiveProps` we prefix with next for next state, so it becomes `searchString` and `nextSearchString`